experiments/sample_ssl4eo.py

```diff
@@ -0,0 +1,265 @@
+""" Sample patch locations for downloading
+
+### run the script:
+## sample new locations with rtree overlap search
+python sample_ssl4eo.py \
+    --save_path ./data \
+    --radius 1320 \
+    --num_cities 10000 \
+    --std 50 \
+    --indices_range 0 250000
+
+## resume from interruption
+python sample_ssl4eo.py \
+    --save_path ./data \
+    --radius 1320 \
+    --num_cities 10000 \
+    --std 50 \
+    --resume ./data/checked_locations.csv \
+    --indices_range 0 250000
+
+### Notes
+# The script will sample locations with rtree overlap search.
+# The script will save the sampled locations to a csv file.
+# By default, GaussianSampler is used to sample locations with a standard deviation of 50 km from top 10000 populated cities. # noqa: E501
+# By default, 25% overlap of adjacent patches is allowed.
+# Radius (meter) is half the wanted patch size.
+
+"""
+
+import argparse
+import csv
+import os
+import time
+import warnings
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+import shapefile
+from rtree import index
+from shapely.geometry import Point, shape
+from torchvision.datasets.utils import download_and_extract_archive
+from tqdm import tqdm
+
+warnings.simplefilter("ignore", UserWarning)
+
+
+""" samplers to get locations of interest points"""
+
+
+class UniformSampler:
+    def sample_point(self) -> List[float]:
+        lon = np.random.uniform(-180, 180)
+        lat = np.random.uniform(-90, 90)
+        return [lon, lat]
+
+
+class GaussianSampler:
```

Comment:
Using a class for this feels like overkill, especially now that we will only define a single sampler. I think you're better off converting each staticmethod to a normal function.